VDA-DID-Registry/contracts/VeridaDIDRegistry.sol

```diff
@@ -5,419 +5,80 @@ pragma solidity ^0.8.6;
-    emit DIDAttributeChanged(identity, name, value, 0, new bytes(0), changed[identity]);
-    changed[identity] = block.number;
-  }
-
-  /**
-   * @notice Revoke an attribute from DID
-   * @dev Check transaction signature and call internal function
-   * @param identity - DID
-   * @param name - attribute name in bytes32 format
-   * @param value - attribute value
-   * @param signature - transaction signature
-   */
-  function revokeAttribute(address identity, bytes32 name, bytes calldata value, bytes calldata signature ) external {
-    {
-      bytes memory unsignedMsg = abi.encodePacked(
-        identity,
-        name,
-        value,
-        nonce[identity]
-      );
-      address owner = identityOwner(identity);
-      require(VeridaDataVerificationLib.validateSignature(unsignedMsg, signature, owner), "Invalid Signature");
-      nonce[identity]++;
+    for (uint i = 0; i < endpoints.length; i++ ) {
+      list.add(endpoints[i]);
     }
-    _revokeAttribute(identity, name, value);
-  }
 
-  /**
-   * @notice Perform 'add' operations of multiple delegates & attributes
-   * @dev Only called after checking the transaction signature
-   * @param identity - DID
-   * @param delegateParams - array of delegates to be added
-   * @param attributeParams - array of attributes to be added
-   */
-  function _bulkAdd(
-    address identity,
-    DelegateParam[] calldata delegateParams,
-    AttributeParam[] calldata attributeParams
-  ) internal {
-    for (uint i = 0; i < delegateParams.length; i++) {
-      _addDelegate(
-        identity, 
-        delegateParams[i].delegateType,
-        delegateParams[i].delegate,
-        delegateParams[i].validity);
-    }
-
-    for (uint i = 0; i < attributeParams.length; i++) {
-      _setAttribute(
-        identity, 
-        attributeParams[i].name,
-        attributeParams[i].value,
-        attributeParams[i].validity,
-        attributeParams[i].proof);
-    }
+    emit Register(didAddress, endpoints);
   }
 
   /**
-   * @notice Perform 'add' operations of multiple delegates & attributes
-   * @dev Check transaction signature and call internal function
-   * @param identity - DID
-   * @param delegateParams - array of delegates to be added
-   * @param attributeParams - array of attributes to be added
-   * @param signature - transaction signature
+   * @dev See {IVeridaDIDRegistry}
    */
-  function bulkAdd(
-    address identity,
-    DelegateParam[] calldata delegateParams,
-    AttributeParam[] calldata attributeParams,
-    bytes calldata signature
-  ) external {
-    {
-      bytes memory unsignedMsg = abi.encodePacked(identity);
-      for (uint i = 0; i < delegateParams.length; i++) {
-        unsignedMsg = abi.encodePacked(
-          unsignedMsg,
-          delegateParams[i].delegateType,
-          delegateParams[i].delegate,
-          delegateParams[i].validity
-        );
-      }
+  function lookup(address didAddress) external view override returns(string[] memory) {
```

Comment:
Need to throw an error if the didAddress not found.

Reply:
Updated.